compiler/scenario-service/protos/scenario_service.proto

```diff
@@ -274,6 +274,14 @@ message ScenarioError {
     Identifier required_interface_id = 3;
   }
 
+  message UpgradeValidationFailed {
+    ContractRef contract_ref = 1;
+    Identifier srcTemplateId =2 ;
+    repeated string signatories = 4;
+    repeated string observers = 5;
+    KeyWithMaintainers key = 6;
```

Comment:
I think it's worth filing an issue for adding the handling of the ensures clause.